service/storage_service.cc

```diff
@@ -6073,12 +6081,25 @@ future<> storage_service::stream_tablet(locator::global_tablet_id tablet) {
         auto range = tmap.get_token_range(tablet.tablet);
         std::optional<locator::tablet_replica> leaving_replica = locator::get_leaving_replica(tinfo, *trinfo);
         locator::tablet_migration_streaming_info streaming_info = get_migration_streaming_info(tm->get_topology(), tinfo, *trinfo);
+        locator::tablet_replica_set read_from{streaming_info.read_from.begin(), streaming_info.read_from.end()};
+        if (trinfo->transition == locator::tablet_transition_kind::rebuild_v2) {
+            // Choose one of the replicas to stream from.
+            auto replica_opt = maybe_get_primary_replica(tablet.tablet, read_from, [&pending_replica = *trinfo->pending_replica] (const auto& tr) {
```

Comment:
Instead of choosing the primary replica, I think we should pick the nearest replica to stream from. E.g., the one in the same rack -> same dc -> remote  dc.

Reply:
Good point. 

@Deexie Note that there is `topology::sort_by_proximity()`